src/evo/providertx.cpp

```diff
@@ -27,7 +27,7 @@ static bool CheckService(const uint256& proTxHash, const ProTx& proTx, CValidati
         return state.DoS(10, false, REJECT_INVALID, "bad-protx-addr");
     }
 
-    int mainnetDefaultPort = Params(CBaseChainParams::MAIN).GetDefaultPort();
+    int mainnetDefaultPort = CreateChainParams(CBaseChainParams::MAIN)->GetDefaultPort();
```

Comment:
This will be quite expensive. Better to have a static variable that holds an instance of it and then access it here.

Reply:
I'm not really sure where that static variable should go.

Reply:
Just above that line.
```
static int mainnetDefaultPort = CreateChainParams(CBaseChainParams::MAIN)->GetDefaultPort();
```

Reply:
Ah no, sorry. Not above that line, but replace that line...actually you just have to make mainnetDefaultPort static

Reply:
https://github.com/dashpay/dash/pull/3007/commits/46dd6a357564301c29c3aecedb404271378d09ad